napari/utils/events/evented_model.py

```diff
@@ -75,3 +76,37 @@ def __setattr__(self, name, value):
     @property
     def events(self):
         return self._events
+
+    def asdict(self):
+        """Convert a model to a dictionary."""
+        warnings.warn(
+            (
+                "The `asdict` method has been renamed `dict` and is now deprecated. It will be"
+                " removed in 0.4.7"
+            ),
+            category=FutureWarning,
+            stacklevel=2,
+        )
+        return self.dict()
+
+    def update(self, values):
+        """Update a model in place.
+
+        Parameters
+        ----------
+        values : dict, napari.utils.events.EventedModel
+            Values to update the model with. If an EventedModel is passed it is first
+            converted to a dictionary. The keys of this dictionary must be found as
+            attributes on the current model.
+        """
+        if isinstance(values, self.__class__) and hasattr(values, 'dict'):
+            values = values.dict()
+        if not isinstance(values, dict):
+            raise ValueError(f"Unsupported update from {type(values)}")
+
+        with self.events.blocker() as block:
+            for key, value in values.items():
+                setattr(self, key, value)
+
+        if block.count:
+            self.events(Event(self))
```

Comment:
Did this code emits all individual signals for these values which were changed? Some of the functions could be connected only to specific names.

Reply:
it's the same as before, i'm pretty sure yes - to block all events you'd use `with self.events.blocker_all():`

Reply:
note this is confirmed by our tests actually, so i know it works

Reply:
ok.